core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -214,6 +215,23 @@ public static void warmUpLoop() {
   }
 
   /**
+   * Waits for a condition to be satisfied until a timeout occurs.
+   *
+   * @param description a description of what causes condition to evaluation to true
+   * @param condition the condition to wait on
+   * @param timeoutMs the number of milliseconds to wait before giving up and throwing an exception
+   */
+  public static void waitFor(String description, Function<Void, Boolean> condition, int timeoutMs) {
+    long start = System.currentTimeMillis();
+    while (!condition.apply(null)) {
+      if (System.currentTimeMillis() - start > timeoutMs) {
+        throw new RuntimeException("Timed out waiting for " + description);
+      }
+      CommonUtils.sleepMs(20);
```

Comment:
Not sure if this should be a constant, but not something this PR needs to address.